crates/uv/src/commands/project/init.rs

```diff
@@ -94,6 +98,26 @@ pub(crate) async fn init(
         }
     }
 
+    // Initialize the version control system.
+    let in_existing_vcs = existing_vcs_repo(path.parent().unwrap_or(&path));
+    let vcs = match (version_control, in_existing_vcs) {
+        (None, false) => VersionControl::default(),
+        (None, true) => VersionControl::None,
+        (Some(vcs), false) => vcs,
+        (Some(vcs), true) => {
+            warn_user!(
+                "The project is already in a version control system, `--vcs {vcs}` is ignored",
+            );
+            VersionControl::None
+        }
+    };
```

Comment:
I feel like if you do `--vcs mercurial` and we're in a git repository we should fail instead of warning and ignoring?